crates/solvers/src/domain/solver/dex/solver.rs

```diff
@@ -54,14 +68,25 @@ impl Dex {
         gas: auction::GasPrice,
     ) -> Option<solution::Solution> {
         let swap = {
-            let order = dex::Order::new(&order);
+            let order = self.partial_fill_handler.dex_order(&order);
```

Comment:
Initially the code only returned the fill amount we should try to for a given order which arguably somewhat fits with fok orders. But because that required a bunch of matching outside the `PartialFillHandler` I decided to manage more or less everything inside the component. Alternatively one could call it something like `FillManager` or `ExecutionManager` which would make it less confusing why we pass it fok orders. WDYT?

Reply:
I think we can call it `fills::Fills` then so its general for all orders.